### ISSUE TITLE
Higher the entity limit

### ISSUE THREAD

**Issue by mcNuggets1** [Facepunch/garrysmod-requests#935]
Maximum 8194 entites is way too less for big servers with 128 players on it.
I would suggest to double the limit.

I mean, nearly everything is an entity. The limit is just reached too easily.

**Comment by Kefta** [comment]
It really isn't. If you have 50 entities associated to each player (unrealistic on its own) and you have 100 map entities, then you're still 500 under the limit. Maybe one of your addons is creating entities that are never freed, and thus your count is going too high? The problem is with increasing it is you're now putting on an extra (numerical) bit in networking and everything has to be changed to reflect that, never mind the extra bloat.

**Comment by mcNuggets1** [comment]
Every single weapon is an entity.
The physgun beam is an entity.
Effects/Explosions are entities.
Viewmodels are entities.
Sprites are entities.
Everysingle prop is one entity.
Constraints are entities.
Every parented entity counts as one entity either.

It is really easy to reach the limit when everything is an entity.

**Comment by neico** [comment]
On that regard I'd like to point to this page: https://developer.valvesoftware.com/wiki/Entity_limit

It explains in further detail how the limit came to be and how it's structured.

**Comment by Rakaesa** [comment]
@Kefta 

Hey. I know this is an old (and often requested) issue so sorry for the necro, but I really do think an entity limit increase would be nice. I've ran a lot of roleplay servers and we've very frequently hit the limits. I've always chalked it up to being "just a source engine problem" but if it really can be increased that would be phenomenal. It would allow for bigger, interesting maps, proper roleplay events without worrying about hitting the limits...If it eventually causes performance issues, leave that up to the server owner to balance, no?

**Comment by Kefta** [comment]
It affects everyone when you raise the limit because now, the max entity edict takes up more bits. It also breaks addons that hardcoded the entity bit count before. These are not so-major of concerns that it kills this proposal - the limit is too low for how most servers operate now, but it is quite late in the game's history for a breaking change and thus I doubt this will ever take priority.

**Comment by aStonedPenguin** [comment]
> it also breaks addons that hardcoded the entity bit count before

I would imagine those aren't particularly common.

Seeing as how it is no longer 2017 and with modern hardware you can feasibly hit the entity limit while still having a stable server this is a no brainier.

**Comment by akotak64** [comment]
> > it also breaks addons that hardcoded the entity bit count before
> 
> I would imagine those aren't particularly common.
> 
> Seeing as how it is no longer 2017 and with modern hardware you can feasibly hit the entity limit while still having a stable server this is a no brainier.

truth

**Comment by Kefta** [comment]
Raise everything, I don't care anymore

**Comment by SirYodaJedi** [comment]
Apparently Jabroni Brawl raised their limit to 65535 edicts (and removed the VBSP entity limit outright); if any engine licensor would know what needs to be raised, it'd be them. ~~also ask them for their func_vehicle code while you're at it~~

As for issues with raising the limit, while I'm sure most computers running gmod that can handle the increased memory requirements caused by bumping the limit to 16k or even 65k, I'd be less sure about networks being able to handle the increased bandwidth (there are some pretty shoddy networks out there). It'd still require a new network protocol, but I think it'd make sense to have a server-side cvar that controls the max edicts, and this gets broadcast as part of the server "header" or whatever that shows in the server list, and the client will know how much to allocate when connecting based upon that (or just always allocate 65k on client, but use the cvar to clamp how much is allowed to be networked).

(also, why is the post before me linking to leaked code?)

**Comment by ynot01** [comment]
I made an addon just now which creates clientside "permaprops" to side-step this issue. Note that these props do not have collision, do not exist server-side, high amounts will cause frame drops, and my code is not battle-tested. When spawning ~7500 I got around 30 fps when looking at them all

https://github.com/ynot01/clientside_permaprops

Feel free to contribute or fork :heart:

**Comment by SirYodaJedi** [comment]
> post by @ynot01 

I'm confused what this achieves that static props do not. Static props already don't count toward towards any entity limits except VBSP's, and are still allowed to have collision.

**Comment by ynot01** [comment]
you are correct on all counts, it's just a convenience thing since servers don't tend to update maps often
